### DISCUSSION TITLE
get apphandle instance

### DISCUSSION THREAD

**Discussion by cosoc** [tauri-apps/tauri#5230] [Q&A] [answered]
I am a rust newbie! tauri confuses me so much! I want to use the apphandle instance in other functions, but I don't know how to easily get the apphandle for sending events!
![image](https://user-images.githubusercontent.com/19447682/190935863-32641958-ee57-4a2e-a9dd-9717417ef5c7.png)

**Comment by FabianLars** [comment, marked as answer]
In the `.setup` hook yuo can get an AppHandle via `let handle = app.handle()`. This needs to happen outside of the listen_global call, otherwise you'll get compiler errors.

For other use-cases you can even inject an AppHandle into tauri commands as shown here https://tauri.app/v1/guides/features/command#accessing-an-apphandle-in-commands
